packages/approval-controller/src/ApprovalController.ts

```diff
@@ -590,7 +627,10 @@ export class ApprovalController extends BaseControllerV2<
       this.reject(id, rejectionError);
     }
     this._origins.clear();
-    this.update(() => getDefaultState());
+    this.update((draftState) => {
+      draftState.pendingApprovals = {};
+      draftState.pendingApprovalCount = 0;
+    });
```

Comment:
This has to do with mobile, since I couldn't find any instances of `clear` within the extension or snaps code.

When requesting a user approval in mobile, `ApprovalController.clear()` is called to remove all pending approvals.

If we don't stop it from clearing approval flows, it will also stop them in between approvals.

Other alternatives include an option in the `requestUserApproval` function to stop calling `clear` at will (defaulting to the current behaviour).